src/requireLocation.ts

```diff
@@ -0,0 +1,10 @@
+export const requireLocation = (store: any) => {
+  return (nextState: any, replace: any) => {
+    if (!store.getState().user.location) {
+      replace({
+        pathname: '/',
+        state: { nextPathname: nextState.location.pathname }
```

Comment:
Oh, yeah we don't need that. Also, by removing that the query param no longer appears in the URL.